src/Compiler/Driver/CompilerImports.fs

```diff
@@ -1172,7 +1172,14 @@ and [<Sealed>] TcImports
         match (Some ccu.Contents, nsname) ||> List.fold matchNameSpace with
         | Some ns ->
             match Map.tryFind tname ns.ModuleOrNamespaceType.TypesByMangledName with
-            | Some _ -> true
+            | Some e ->
+                if publicOnly then
+                    match e.TypeReprInfo with
+                    | TILObjectRepr data ->
+                        let (TILObjectReprData(_, _, tyDef)) = data
+                        tyDef.Access = ILTypeDefAccess.Public
```

Comment:
Should this be `AccessibleFromEverywhere`, `AccessibleFromSomeFSharpCode` or `AccessibleFromSomewhere`? I.e. do we care about internals + `InternalsVisibleTo`? Probably not for `IsReadOnlyAttribute` in particular, but maybe for others?

Reply:
No ... remember this is system types I.e framework types, and the scenarios are marker types, ie ones whose names signal behaviour, so duplicating them is more right than demanding internal visibility.